gomock/callset.go

```diff
@@ -87,8 +87,13 @@ func (cs callSet) FindMatch(receiver interface{}, method string, args []interfac
 		}
 	}
 
-	if len(expected)+len(exhausted) == 0 {
+	expectedNum := len(expected)
+	exhaustedNum := len(exhausted)
+
+	if expectedNum == 0 && exhaustedNum == 0 {
 		fmt.Fprintf(&callsErrors, "there are no expected calls of the method %q for that receiver", method)
+	} else if exhaustedNum > 0 && callsErrors.Len() == 0 {
```

Comment:
I think a comment here explaining what error scenario we are handling (i.e. an ordering error) and the rational behind the check for this scenario would be really good.